api/src/domain/user/encryptTaxIdFactory.ts

```diff
@@ -0,0 +1,22 @@
+import { UserData } from "@shared/userData";
+import { EncryptionDecryptionClient, EncryptTaxId } from "../types";
+import { maskTaxId } from "./maskTaxId";
+
+export const encryptTaxIdFactory = (encryptionDecryptionClient: EncryptionDecryptionClient): EncryptTaxId => {
+  return async (userData: UserData): Promise<UserData> => {
+    const maskedTaxId = maskTaxId(userData.profileData.taxId as string);
+    const encryptedTaxId = await encryptionDecryptionClient.encryptValue(
+      userData.profileData.taxId as string
+    );
+
+    const user: UserData = {
+      ...userData,
+      profileData: {
+        ...userData.profileData,
+        taxId: maskedTaxId,
+        encryptedTaxId: encryptedTaxId,
+      },
+    };
+    return user;
+  };
```

Comment:
could we make use of this code in userRouter also? so that we're not duplicating code